code/frameworks/nextjs/src/images/context.ts

```diff
@@ -0,0 +1,16 @@
+import { createContext } from 'react';
+import type { ImageProps, StaticImageData } from 'next/image';
+import type { ImageProps as LegacyImageProps } from 'next/legacy/image';
+
+// StaticRequire needs to be in scope for the TypeScript compiler to work.
+// See: https://github.com/microsoft/TypeScript/issues/5711
+// Since next/image doesn't export StaticRequire we need to re-define it here and set src's type to it.
```

Comment:
https://app.circleci.com/pipelines/github/storybookjs/storybook/46783/workflows/1af13e69-d40a-4c2a-8a31-1bb3ad3c8401/jobs/511829

I've never seen this error before but putting StaticRequire in the scope fixed it.